homeassistant/components/fibaro/sensor.py

```diff
@@ -80,112 +106,73 @@ async def async_setup_entry(
 ) -> None:
     """Set up the Fibaro controller devices."""
     entities: list[SensorEntity] = []
+
     for device in hass.data[DOMAIN][entry.entry_id][FIBARO_DEVICES][Platform.SENSOR]:
-        entities.append(FibaroSensor(device))
+        entity_description = None
+        for desc in MAIN_SENSOR_TYPES:
+            if desc.key == device.type:
+                entity_description = desc
+                break
+        # main sensors are created even if the entity type is not known
+        entities.append(FibaroSensor(device, entity_description))
+
     for platform in (Platform.COVER, Platform.LIGHT, Platform.SENSOR, Platform.SWITCH):
         for device in hass.data[DOMAIN][entry.entry_id][FIBARO_DEVICES][platform]:
-            if "energy" in device.interfaces:
-                entities.append(FibaroEnergySensor(device))
-            if "power" in device.interfaces:
-                entities.append(FibaroPowerSensor(device))
+            for entity_description in ADDITIONAL_SENSOR_TYPES:
+                if entity_description.key in device.properties:
```

Comment:
Instead of this loop, can we make `ADDITIONAL_SENSOR_TYPES` a dict?

Reply:
This makes no sense for `ADDITIONAL_SENSOR_TYPES` as we can have more than one additional sensor per device, so in my opinion a loop is anyway needed.